### ISSUE TITLE
Calling `curl_global_init` after `curl_global_cleanup` failed

### ISSUE THREAD

**Issue by LinZhihao-723** [curl/curl#13805]
### I did this

In our project, we designed the curl lib to be loaded using `curl_global_init` where it can be deinit by calling `curl_global_cleanup`. If later we still need to use curl in the same process, we will call `curl_global_init` again. This "calling `curl_global_init` after `curl_global_cleanup`" would cause an error when I try to use the easy handle since the second call on macOS. Here is a sample code to reproduce the problem:
```
#include <iostream>
#include <fstream>
#include <curl/curl.h>
#include <chrono>
#include <thread>

size_t write_data(void *ptr, size_t size, size_t nmemb, std::ofstream* stream) {
    stream->write((char*) ptr, size * nmemb);
    return size * nmemb;
}

void download(std::string const& filename) {
    curl_global_init(CURL_GLOBAL_ALL);
    std::string url =
        "https://raw.githubusercontent.com/y-scope/clp/main/components/core/"
        "tests/test_network_reader_src/random.log";
    std::ofstream file;
    CURLcode res;
    CURL* curl = curl_easy_init();
    file.open(filename, std::ios::binary);
    if (!file.is_open()) {
        std::cerr << "Cannot open file for writing" << std::endl;
        curl_easy_cleanup(curl);
        curl_global_cleanup();
        return;
    }

    curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, write_data);
    curl_easy_setopt(curl, CURLOPT_WRITEDATA, &file);
    res = curl_easy_perform(curl);

    if (res != CURLE_OK) {
        std::cerr << "curl_easy_perform() failed: " << curl_easy_strerror(res) << std::endl;
        curl_easy_cleanup(curl);
        curl_global_cleanup();
        return;
    }

    curl_easy_cleanup(curl);
    std::this_thread::sleep_for(std::chrono::seconds(3));
    curl_global_cleanup();

    file.close();
}

int main() {
    download("file1.log");
    download("file2.log");
    return 0;
}
```
`download` is called twice, each call will call `curl_global_init` and `curl_global_deinit`. The first call will succeed but the second call will fail. Notice that this problem only happens on MacOS. It's not appearing on Ubuntu (neither 20.04 nor 22.04)

Error message:
```
curl_easy_perform() failed: SSL connect error
```

### I expected the following

In the example above, two `download` calls should both success.

### curl/libcurl version

8.4.0

### operating system

MacOS 14.2.1

**Comment by vszakats** [comment]
Reported earlier in #12525.

**Comment by bagder** [comment]
Fixed in 8.6.0
